version/src/lib.rs

```diff
@@ -48,10 +47,8 @@ pub mod error {
         /// Cannot encode unsupported version from Parity SCALE to JSON
         UnsupportedScaleEncode,
         /// JSON (de)serialization issue
-        #[cfg(feature = "json")]
         Serde,
         /// Parity SCALE (de)serialization issue
-        #[cfg(feature = "scale")]
```

Comment:
@s8sato removed your comment accidentally. The comment said that this was removed to prevent structural inconsistencies between client and server as we've seen in #2111 

Good thinking

Reply:
> Removed these attributes for the same reason as `#[cfg(feature = "role")]`, namely to fix the serialized format of the enum